.github/workflows/version-bump.yml

```diff
@@ -0,0 +1,117 @@
+        with:
+          node-version: 16.x
+          registry-url: https://registry.npmjs.org/ # Needed for auth
+      - name: cache all node_modules
+        id: cache-modules
+        uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-node_modules-${{ hashFiles('yarn.lock', '**/package.json') }}
+      - name: find location of global yarn cache
+        id: yarn-cache
+        if: steps.cache-modules.outputs.cache-hit != 'true'
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - name: cache global yarn cache
+        uses: actions/cache@v2
+        if: steps.cache-modules.outputs.cache-hit != 'true'
+        with:
+          path: ${{ steps.yarn-cache.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      - name: yarn install
+        run: yarn install --frozen-lockfile
+      # End of yarn setup
+
+      - name: 'Set release name'
+        id: set_release_name
+        run: node scripts/set-release-name.js
+      - name: 'Configure git'
+        run: |
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config --global user.name "github-actions[bot]"
+      - name: 'Create topic branch'
+        run: |
+          git config --global --add safe.directory "$GITHUB_WORKSPACE"
+          BRANCH=$(git branch --list topic/v${{ steps.set_release_name.outputs.release_version }})
+          if [ ! -z "$BRANCH" ]; then
+              git branch -D topic/v${{ steps.set_release_name.outputs.release_version }}
+          fi
+          git checkout -b topic/v${{ steps.set_release_name.outputs.release_version }}
+      - name: Run backstage-cli versions:bump --release ${{ inputs.release_line || 'next' }} command
+        run: yarn backstage-cli versions:bump --release ${{ inputs.release_line || 'next' }}
+      - name: 'Check for changes'
+        id: check_for_changes
+        run: |
+          CHANGES=$(git status --porcelain)
+          if [ -z "$CHANGES" ]; then
+              echo "No changes"
+              echo "HAS_CHANGES=0" >> $GITHUB_OUTPUT
+          else
+              echo "Has changes"
+              echo "HAS_CHANGES=1" >> $GITHUB_OUTPUT
+          fi
+      - name: 'Commit changes'
+        if: ${{ steps.check_for_changes.outputs.HAS_CHANGES == 1 }}
+        run: |
+          git status
+          git add -A -- :!.npmrc
+          git commit -m "v${{ steps.set_release_name.outputs.release_version }} version bump"
+          git push origin topic/v${{ steps.set_release_name.outputs.release_version }}
+      - name: 'Create Pull Request'
+        if: ${{ steps.check_for_changes.outputs.HAS_CHANGES == 1 }}
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{secrets.VERSION_BUMP_TOKEN}}
+          script: |
+            await github.rest.pulls.create({
```

Comment:
assuming this does not create more than one PR for every release if we run this in cron? 😅

Reply:
As long as there's no changes then it won't. This is driven by the "Check for changes" step at line 76 which basically check to see if there are any changes using a git command.